component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/record/AvroSchema.java

```diff
@@ -184,7 +185,28 @@ private Entry fromAvro(final Field field) {
         return AvroSchema.buildFromAvro(field, fieldType, fieldSchema);
     }
 
+    /**
+     * Builds a TCK {@link org.talend.sdk.component.api.record.Schema.Entry} from an Avro {@link Field}.
+     * <p>
+     * Entry original name is stored in an Avro property (named {@link KeysForAvroProperty#LABEL}), among
+     * other possible properties. As this property is an implementation detail of this Avro TCK implementation,
+     * it should never leak in the resulting {@link org.talend.sdk.component.api.record.Schema.Entry} props.
+     * <p>
+     * <i>Note regarding the warning suppression: ideally, one should avoid using the {@link Field#getProps()}</i>
+     * method as it is deprecated...
+     */
+    @SuppressWarnings("deprecation")
     private static Entry buildFromAvro(final Field field, final Type type, final AvroSchema elementSchema) {
+        Map<String, String> tckEntryProps;
+
+        if (field.getProps().containsKey(KeysForAvroProperty.LABEL)) {
+            tckEntryProps = new HashMap<>(field.getProps());
+            tckEntryProps.remove(KeysForAvroProperty.LABEL);
```

Comment:
As discussed on slack, we shouldn't remove this property from others.

Reply:
I still think this is a bug.

However, I'll close the PR as we agreed we won't (can't?) fix it in a near future.